src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -262,10 +263,25 @@ protected override void ContextUnloaded(ExecutionContext context)
             base.ContextUnloaded(context);
             if (Diagnostic is not null)
                 currentNodeOfInvocationTree = currentNodeOfInvocationTree.Parent;
-            if (!contractTasks.Remove(context, out var awaiter)) return;
-            if (UncaughtException is not null)
-                throw new VMUnhandledException(UncaughtException);
-            awaiter.SetResult(this);
+            if (context.Script != CurrentContext?.Script)
```

Comment:
Do it before ` base.ContextUnloaded(context);` in order to avoid the null?
https://github.com/neo-project/neo-vm/blob/1d8995d33640e1a907341ea380cdbc5ce217e3c1/src/neo-vm/ExecutionEngine.cs#L112

Reply:
Before that `context` is `CurrentContext`.